libs/coin-modules/coin-solana/src/prepareTransaction.ts

```diff
@@ -837,4 +841,23 @@ function validateRecipientRequiredMemo(
   }
 }
 
+function isValidCommandDescriptor(commandDescriptor: CommandDescriptor | undefined) {
+  const txCommand = commandDescriptor?.command as StakeCreateAccountCommand | undefined;
+
+  if (
+    commandDescriptor &&
+    txCommand?.amount &&
+    txCommand.stakeAccRentExemptAmount &&
+    txCommand.fromAccAddress &&
+    txCommand.stakeAccAddress &&
+    txCommand.delegate &&
+    txCommand.seed &&
+    commandDescriptor.fee
+  ) {
+    return true;
+  }
+
+  return false;
+}
```

Comment:
I don't understand why we need all this checks. It is not like `txCommand` is `unknown`, or `any`.

How about

```suggestion
function isStakeCreateAccountCommandDescriptor(commandDescriptor: CommandDescriptor | undefined): commandDescriptor is CommandDescriptor & { command: StakeCreateAccountCommand } {
  return !!commandDescriptor && commandDescriptor.command.kind === "stake.createAccount"
}
```

Reply:
Thanks for the feedback! We need all these checks to **ensure** that `commandDescriptor` **contains** all the necessary data so we **don't** have to generate a new one. For instance, on the LLD side, the command descriptor **arrives** without an amount. If we **don't** check it, the transaction won't be properly prepared.

Reply:
Ok thanks for the explanation. You should paybe add a comment, since this is really not ovious in the first place.

The predicate part (`commandDescriptor is CommandDescriptor`) it still relevant though.